code/game/objects/effects/spawners/bombspawner.dm

```diff
@@ -6,52 +6,118 @@
 	name = "bomb"
 	icon = 'icons/hud/screen_gen.dmi'
 	icon_state = "x"
-	/// The initial temperature of the plasma tank.
-	var/temp_p = 1413
-	/// The initial temperature of the oxygen tank.
-	var/temp_o = 141.3
-	/// The initial pressure of the plasma tank.
-	var/pressure_p = TANK_LEAK_PRESSURE - 1
-	/// The initial pressure of the oxygen tank.
-	var/pressure_o = TANK_LEAK_PRESSURE - 1
-	/// The typepath of the assembly to attach to the TTV.
-	var/assembly_type
-
-/obj/effect/spawner/newbomb/Initialize(mapload)
+	/* Gasmixes for tank_one and tank_two of the ttv respectively. 
+	 * Populated on /obj/effect/spawner/newbomb/Initialize, depopulated right after by the children procs.
+	 */
+	var/datum/gas_mixture/first_gasmix
+	var/datum/gas_mixture/second_gasmix
+
+/** 
+ * The part of code that actually spawns the bomb. Always call the parent's initialize first for subtypes of these.
+ *
+ * Arguments: 
+ * * assembly - An assembly typepath to add to the ttv.
+ */
+/obj/effect/spawner/newbomb/Initialize(mapload, assembly = NONE)
```

Comment:
```suggestion
/obj/effect/spawner/newbomb/Initialize(mapload, assembly = null)
```
`NONE` refers to bitflags, just a clarity issue.
In here we expect this argument to be typed as a nullable type path, right?

Reply:
Yep, my mistake.